libraries/ktx/src/ktx/Reader.cpp

```
@@ -0,0 +1,186 @@
+//
+//  Reader.cpp
+//  ktx/src/ktx
+//
+//  Created by Zach Pomerantz on 2/08/2017.
+//  Copyright 2017 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+#include "KTX.h"
+
+#include <list>
+#include <QtGlobal>
+
+namespace ktx {
+    class ReaderException: public std::exception {
+    public:
+        ReaderException(std::string explanation) : _explanation(explanation) {}
+        const char* what() const override {
+            return ("KTX deserialization error: " + _explanation).c_str();
+        }
+    private:
+        std::string _explanation;
+    };
+
+    bool checkEndianness(uint32_t endianness, bool& matching) {
+        switch (endianness) {
+        case Header::ENDIAN_TEST: {
+            matching = true;
+            return true;
+            }
+            break;
+        case Header::REVERSE_ENDIAN_TEST:
+            {
+                matching = false;
+                return true;
+            }
+            break;
+        default:
+            return false;
```

this should throw an exception.  If the endianness check doesn't match either value, then the file is invalid.